core/src/main/java/org/apache/calcite/sql/SqlCharStringLiteral.java

```diff
@@ -77,8 +77,13 @@ private NlsString getValueNonNull() {
       String stringValue = nlsString.getValue();
       writer.literal(
           writer.getDialect().quoteStringLiteral(stringValue));
+    } else if (!writer.toString().contains("UNPIVOT") && writer.toString().contains("PIVOT")
+        && (writer.toString().endsWith("FOR")
+        || (writer.toString().contains("IN") && writer.toString().endsWith("AS")))) {
+      writer.literal(nlsString.asSql(true, true, writer.getDialect()).replaceAll("'", ""));
```

Comment:
this code can be avoided
if its done for removing from the axiscolumn , create it as a SqlIdentifier rather than SqlCharStringLiteral

Reply:
resolved